tests/unit/processor/key_checker/test_key_checker_rule.py

```diff
@@ -0,0 +1,167 @@
+            ),
+        ],
+    )
+    def test_create_from_dict_validates_config(self, rule, error, message):
+        if error:
+            with pytest.raises(error, match=message):
+                KeyCheckerRule._create_from_dict(rule)
+        else:
+            keychecker_rule = KeyCheckerRule._create_from_dict(rule)
+            assert hasattr(keychecker_rule, "_config")
+            for key, value in rule.get("key_checker").items():
+                assert hasattr(keychecker_rule._config, key)
+                sad = type(value)
+                temp_list = list(getattr(keychecker_rule._config, key))
+                temp_list.sort()
+                if isinstance(value, list):
+                    temp_list = list(getattr(keychecker_rule._config, key))
+                    temp_list.sort()
+                    value.sort()
+                    assert value == temp_list
+                else:
+                    assert value == getattr(keychecker_rule._config, key)
+
+    @pytest.mark.parametrize(
+        ["testcase", "rule1", "rule2", "equality"],
+        [
+            (
+                "should not be equal, because the name is different",
+                {
+                    "filter": "*",
+                    "key_checker": {
+                        "key_list": ["key1"],
+                        "output_field": "missing_fields",
+                    },
+                },
+                {
+                    "filter": "*",
+                    "key_checker": {
+                        "key_list": ["key2"],
+                        "output_field": "missing_fields",
+                    },
+                },
+                False,
+            ),
+            (
+                "should be equal, because only the order has changed",
+                {
+                    "filter": "*",
+                    "key_checker": {
+                        "key_list": ["key1", "key2"],
+                        "output_field": "missing_fields",
+                    },
+                },
+                {
+                    "filter": "*",
+                    "key_checker": {
+                        "key_list": ["key2", "key1"],
+                        "output_field": "missing_fields",
+                    },
+                },
+                True,
+            ),
+            (
+                "should not be equal, because the keys are different",
+                {
+                    "filter": "*",
+                    "key_checker": {
+                        "key_list": ["key1.key2"],
+                        "output_field": "missing_fields",
+                    },
+                },
+                {
+                    "filter": "*",
+                    "key_checker": {
+                        "key_list": ["key1"],
+                        "output_field": "missing_fields",
+                    },
+                },
+                False,
+            ),
```

Comment:
```suggestion
            ),
                        (
                "should be equal, because unique keys are the same",
                {
                    "filter": "*",
                    "key_checker": {
                        "key_list": ["key1", "key2", "key2"],
                        "output_field": "missing_fields",
                    },
                },
                {
                    "filter": "*",
                    "key_checker": {
                        "key_list": ["key2", "key1", "key1"],
                        "output_field": "missing_fields",
                    },
                },
                True,
            ),
```
please add this test